### ISSUE TITLE
Support --scope flag in `lerna bootstap` command

### ISSUE THREAD

**Issue by lukebatchelor** [lerna/lerna#383]
It currently supports the --ignore flag and we'll be using that to work around this at the moment, but it would be nice for consistency and clean-ness if we could get this into lerna as well.

Will be working on the quick work around script tomorrow, if this is something that we would like in lerna I can start on the PR after that. If not, we'll just stick with the work around.

**Comment by seansfkelley** [comment]
This [already exists in Asini](https://github.com/asini/asini/pull/8) if you want to port it from there, or just switch to using it.

**Comment by thejameskyle** [comment]
Sigh, they really are shooting themselves in the foot working on a separate project. Could've improved that PR so much.

**Comment by seansfkelley** [comment]
How? I'm curious. It seems like a pretty straightforward change to promote the scope/ignore resolution to the top level and have everything inherit from it.

**Comment by thejameskyle** [comment]
Just structurally it makes sense to rethink some things about what "filtering" means.

**Comment by lukebatchelor** [comment]
So, I'm looking at pulling in those changes but I want to keep the original authorship. I'm assuming the process is to add asini as a remote, pull its master and cherry pick the commits I want. 

How will this work with the `bin/asini.js` change? That needs to be in `bin/lerna.js` obviously.. Do I just manually do that change for that file?

Also, which branch should I be branching from/merging to? 1.x or master?

Thanks 👍

**Comment by lukebatchelor** [comment]
Got pulled away on a couple of things yesterday but looking at this today.

I believe there might be an error in the asini implementation, could anyone verify this for me?

https://github.com/asini/asini/pull/8/files#diff-b8678a10e51138ee78e70339fa5132f4L73

So, the logic for ignoring based on the config files was moved out of PublishCommand and BootstrapCommand and into the Repository class.

```
export default class Repository {
  /** ... **/
  buildPackageGraph({scope, ignore}) {
    /** ... **/
    if (this.constructor.name === "BootstrapCommand") {
      ignore = ignore || this.bootstrapConfig.ignore;
    }
    if (this.constructor.name === "PublishCommand") {
      ignore = ignore || this.publishConfig.ignore;
    }
    /** ... **/
  }
}
```

If I understand correctly, this.constructor.name will never be "BootstrapCommand" or "PublishCommand" because neither of them extend `Repository`, they extend `Command`.

Is this correct?

I've moved the logic for this into `runPreparations` of `Command` for now.

**Comment by gigabo** [comment]
Hi @lukebatchelor yep, that was a bug.  Nice catch!  It was resolved in https://github.com/asini/asini/pull/15.

Probably easier to just put together a targeted patch to support `--scope` in bootstrap rather than try to start pulling stuff over from Asini.

**Comment by lukebatchelor** [comment]
@gigabo Yea, thats what I've done. I wasnt sure what the "right" thing to do in terms of trying to keep the original authorship. :/

I'm adding a bit more test coverage for this as well, accross the other commands, should be super easy to pull them into asini when its ready if you like. 👍

**Comment by thejameskyle** [comment]
Moving to #386

**Comment by gigabo** [comment]
This landed in [`lerna@2.0.0-beta.31`](https://github.com/lerna/lerna/releases/tag/v2.0.0-beta.31)! 🎉
